cartoframes/context.py

```diff
@@ -234,16 +233,76 @@ def read(self, table_name, limit=None, index='cartodb_id',
         # choose schema (default user - org or standalone - or shared)
         schema = 'public' if not self.is_org else (
             shared_user or self.creds.username())
-        query = 'SELECT * FROM "{schema}"."{table_name}"'.format(
-            table_name=table_name,
-            schema=schema)
+
+        source = '"{schema}"."{table_name}"'.format(table_name=table_name, schema=schema)
+
         if limit is not None:
             if isinstance(limit, int) and (limit >= 0):
-                query += ' LIMIT {limit}'.format(limit=limit)
+                source = '(SELECT * FROM {source} LIMIT {limit})'.format(source=source, limit=limit)
             else:
                 raise ValueError("`limit` parameter must an integer >= 0")
 
-        return self.query(query, decode_geom=decode_geom)
+        query = 'COPY {source} TO stdout WITH (FORMAT csv, HEADER true)'.format(source=source)
+
+        result = self.copy_client.copyto_stream(query)
+        df = pd.read_csv(result)
+
+        return self._clean_DataFrame_from_carto(df, table_name, schema, decode_geom)
+
+    def _clean_DataFrame_from_carto(self, df, table_name, schema='public', decode_geom=False):
+        """
+        Clean a DataFrame with a dataset from CARTO:
+        - remove the_geom_webmercator
+        - use cartodb_id as DataFrame index
+        - process date columns
+        - decode geom
+
+        :param df: DataFrame with a dataset from CARTO
```

Comment:
In catoframes repo we use the google style docstrings:
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

E.g.,:
```df (pandas.DataFrame): DataFrame with a dataset from CARTO```

Reply:
Thanks